service/src/test/java/uk/gov/hmcts/reform/fpl/model/CaseDataTest.java

```diff
@@ -905,6 +908,49 @@ void hearingBookingShouldNotBePresentWhenHearingBookingCaseManagementOrderDoesNo
         }
     }
 
+    @Nested
+    class BuildJudicialMessageList {
+        @Test
+        void shouldBuildDynamicJudicialMessageListFromJudicialMessages() {
+            List<Element<JudicialMessage>> judicialMessages = List.of(
+                element(buildJudicialMessage("Low", "11 November 2020", YES)),
+                element(buildJudicialMessage("Medium", "12 November 2020", NO)),
+                element(buildJudicialMessage("High", "13 November 2020", YES)));
+
+            CaseData caseData = CaseData.builder().judicialMessages(judicialMessages).build();
+            DynamicList expectedDynamicList = ElementUtils
+                .asDynamicList(judicialMessages, null, JudicialMessage::toLabel);
+
+            assertThat(caseData.buildJudicialMessageDynamicList())
+                .isEqualTo(expectedDynamicList);
+        }
+
+        @Test
+        void shouldBuildDynamicJudicialMessageListWithSelectorPropertyFromJudicialMessage() {
+            UUID selectedMessageId = randomUUID();
+
+            List<Element<JudicialMessage>> judicialMessages = List.of(
+                element(buildJudicialMessage("Low", "11 November 2020", YES)),
+                element(buildJudicialMessage("Medium", "12 November 2020", NO)),
+                element(selectedMessageId, buildJudicialMessage("High", "13 November 2020", YES))
+            );
+
+            CaseData caseData = CaseData.builder().judicialMessages(judicialMessages).build();
+            DynamicList expectedDynamicList = ElementUtils
```

Comment:
Thinking the same for this one as well, you are essentially doing `assertThat(asDynamicList(judicialMessages, selected, JudicialMessage::toLabel)).isEqualTo(asDynamicList(judicialMessages, selectedMessageId, JudicialMessage::toLabel))`